src/test/unit/engine/InputTest.cpp

```diff
@@ -165,6 +169,803 @@ TEST(InputTest, Links)
+    "lateralIn",  // destInput
+    1 //propagationDelay
+  );
+
+
+  // Initialize the network
+  net.initialize();
+
+  UInt64* r1OutBuf = (UInt64*)(r1->getOutput("out")->getData().getBuffer());
+  UInt64* r3OutBuf = (UInt64*)(r3->getOutput("out")->getData().getBuffer());
+
+  /* ITERATION #1 with all phases enabled */
+  net.run(1);
+
+  // Validate R1
+  ASSERT_EQ(0u, r1OutBuf[1]); // feedbackIn from R3; delay=1
+  ASSERT_EQ(1u, r1OutBuf[0]); // out (1 + feedbackIn)
+
+  // Validate R3
+  ASSERT_EQ(1u, r3OutBuf[1]); // feedForwardIn from R1; delay=0
+  ASSERT_EQ(0u, r3OutBuf[2]); // lateralIn loopback from R3; delay=1
+  ASSERT_EQ(1u, r3OutBuf[0]); // out (feedForwardIn + lateralIn)
+
+
+  /* Disable Phase 1, containing R1 */
+  net.setMinEnabledPhase(2);
+
+  /* ITERATION #2 with Phase 1 disabled */
+  net.run(1);
+
+  // Validate R1 (it's in a disabled phase, so should be stuck at prior values)
+  ASSERT_EQ(0u, r1OutBuf[1]); // feedbackIn
+  ASSERT_EQ(1u, r1OutBuf[0]); // out
+
+  // Validate R3
+  ASSERT_EQ(1u, r3OutBuf[1]); // feedForwardIn from R1; delay=0
+  ASSERT_EQ(1u, r3OutBuf[2]); // lateralIn loopback from R3; delay=1
+  ASSERT_EQ(2u, r3OutBuf[0]); // out (feedForwardIn + lateralIn)
+
+
+  /* ITERATION #3 with Phase 1 disabled */
+  net.run(1);
+
+  // Validate R1 (it's in a disabled phase, so should be stuck at prior values)
+  ASSERT_EQ(0u, r1OutBuf[1]); // feedbackIn
+  ASSERT_EQ(1u, r1OutBuf[0]); // out
+
+  // Validate R3
+  ASSERT_EQ(1u, r3OutBuf[1]); // feedForwardIn from R1; delay=0
+  ASSERT_EQ(2u, r3OutBuf[2]); // lateralIn loopback from R3; delay=1
+  ASSERT_EQ(3u, r3OutBuf[0]); // out (feedForwardIn + lateralIn)
+
+
+  /* Enable Phase 1, containing R1 */
+  net.setMinEnabledPhase(1);
+
+  /* ITERATION #4 with all phases enabled */
+  net.run(1);
+
+  // Validate R1
+  ASSERT_EQ(3u, r1OutBuf[1]); // feedbackIn from R3; delay=1
+  ASSERT_EQ(4u, r1OutBuf[0]); // out (1 + feedbackIn)
+
+  // Validate R3
+  ASSERT_EQ(4u, r3OutBuf[1]); // feedForwardIn from R1; delay=0
+  ASSERT_EQ(3u, r3OutBuf[2]); // lateralIn loopback from R3; delay=1
+  ASSERT_EQ(7u, r3OutBuf[0]); // out (feedForwardIn + lateralIn)
+
+  /* ITERATION #5 with all phases enabled */
+  net.run(1);
+
+  // Validate R1
+  ASSERT_EQ(7u, r1OutBuf[1]); // feedbackIn from R3; delay=1
+  ASSERT_EQ(8u, r1OutBuf[0]); // out (1 + feedbackIn)
+
+  // Validate R3
+  ASSERT_EQ(8u, r3OutBuf[1]); // feedForwardIn from R1; delay=0
+  ASSERT_EQ(7u, r3OutBuf[2]); // lateralIn loopback from R3; delay=1
+  ASSERT_EQ(15u, r3OutBuf[0]); // out (feedForwardIn + lateralIn)
+
+}
+
+
+TEST(InputTest, SingleL4RegionWithDelayedLoopbackInAndPhaseOnOffOn)
+{
+  // Validates processing of outgoing/incoming delayed link in the context of a
+  // region within a dispabled phase.
+  //
+  // This test simulates a network with a single L4 region, structured as
+  // follows:
+  // o R1 ("L4") is in phase 1
+  // o feedback link with delay=1 from R1 to R1 (loopback)
```

Comment:
It's better to call it lateral link rather than feedback link since there is only one region.